packages/reqresp/src/request/errors.ts

```diff
@@ -73,6 +75,19 @@ export class RequestError extends LodestarError<RequestErrorType> {
  */
 export function responseStatusErrorToRequestError(e: ResponseError): RequestErrorType {
   const {errorMessage, status} = e;
+  // rate limited error from clients have different status, for example: lighthouse responds with 139, teku responds with 1
+  // but all of them has "rate limit" in the error message
+  // refer to https://github.com/ChainSafe/lodestar/issues/8065#issuecomment-3157266196
+  const errorMessageLowercase = errorMessage.toLowerCase();
+  if (errorMessageLowercase.includes("rate limit")) {
+    return {code: RequestErrorCode.RESP_RATE_LIMITED};
+  }
+
+  // Grandine may return this without standard RespStatus, see https://github.com/ChainSafe/lodestar/issues/8110
+  if (errorMessageLowercase.includes("wait ")) {
+    return {code: RequestErrorCode.RESP_TIMEOUT};
+  }
+
   switch (status) {
```

Comment:
Can we also add a RespStatus.RATE_LIMITED (which == 139) and check in this switch?
This will help in case error messages in clients change.